engine/execution/checker/engine.go

```diff
@@ -0,0 +1,143 @@
+package checker
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/consensus/hotstuff/model"
+	"github.com/onflow/flow-go/consensus/hotstuff/notifications"
+	"github.com/onflow/flow-go/engine"
+	"github.com/onflow/flow-go/engine/execution/state"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/state/protocol"
+	"github.com/onflow/flow-go/storage"
+	"github.com/onflow/flow-go/utils/logging"
+)
+
+type Engine struct {
+	notifications.NoopConsumer // satisfy the FinalizationConsumer interface
+
+	unit      *engine.Unit
+	log       zerolog.Logger
+	state     protocol.State
+	execState state.ExecutionState
+	seals     storage.Seals // to compare the executed result with the sealed result
+}
+
+func New(
+	logger zerolog.Logger,
+	state protocol.State,
+	execState state.ExecutionState,
+	seals storage.Seals,
+) *Engine {
+	return &Engine{
+		unit:      engine.NewUnit(),
+		log:       logger.With().Str("engine", "checker").Logger(),
+		state:     state,
+		execState: execState,
+		seals:     seals,
+	}
+}
+
+func (e *Engine) Ready() <-chan struct{} {
+	return e.unit.Ready()
+}
+
+func (e *Engine) Done() <-chan struct{} {
+	return e.unit.Done()
+}
+
+func (e *Engine) SubmitLocal(event interface{})                      {}
+func (e *Engine) Submit(originID flow.Identifier, event interface{}) {}
+func (e *Engine) ProcessLocal(event interface{}) error {
+	return fmt.Errorf("check don't process message")
+}
+func (e *Engine) Process(originID flow.Identifier, event interface{}) error {
+	return fmt.Errorf("check don't process message")
+}
+
+// when a block is finalized check if the last sealed has been executed,
+// if it has been executed, check whether if the sealed result is consistent
+// with the executed result
+func (e *Engine) OnFinalizedBlock(block *model.Block) {
+	err := e.checkExecutionConsistency()
+	if err != nil {
+		e.log.Fatal().Err(err).Msg("execution consistency check failed")
+	}
+}
+
+// CheckExecutionConsistencyWithBlock when a block is executed, if the block has been sealed, check whether if the sealed result is consistent
+// with the executed result
+func (e *Engine) CheckExecutionConsistencyWithBlock(block *flow.Block, mycommit flow.StateCommitment) {
+	err := e.checkExecutedBlockConsistency(block, mycommit)
+	if err != nil {
+		e.log.Fatal().Err(err).
```

Comment:
Won't crashing a node casue an infinite loop? Container will restart and we are back to the same situation.
Shouldn't we rather have some idle-mode, so keep the node alive, but doing essentially nothing?

Reply:
Good point. 

However, I was afraid the alive node is still able to respond to receipt requests from consensus.